apps/bfd-data-npi/src/main/java/gov/cms/bfd/data/npi/lookup/NPIOrgLookup.java

```diff
@@ -94,37 +103,40 @@ public Optional<String> retrieveNPIOrgDisplay(Optional<String> npiNumber) {
    * Reads all the npi number and npi org data fields from the NPPES file which was downloaded
    * during the build process.
    *
-   * <p>See {@link gov.cms.bfd.server.war.NPIDataUtilityApp} for details.
-   *
-   * @param includeFakeNPIOrgCode whether to include the fake testing NPI Org
+   * @param inputStream for the npi file
+   * @throws IOException if there is an issue reading file
+   * @return the hashmapped for npis and the npi org names
    */
-  private void readNPIOrgDataFile(boolean includeFakeNPIOrgCode) throws IOException {
-    final InputStream npiOrgStream =
-        App.class.getClassLoader().getResourceAsStream(App.NPI_RESOURCE);
-
-    if (npiOrgStream != null) {
-      try (final BufferedReader npiOrgIn =
-          new BufferedReader(new InputStreamReader(npiOrgStream))) {
-        String line = "";
-        while ((line = npiOrgIn.readLine()) != null) {
-          String npiOrgColumns[] = line.split("\t");
-          if (npiOrgColumns.length == 2) {
-            npiOrgHashMap.put(
-                npiOrgColumns[0].replace("\"", ""), npiOrgColumns[1].replace("\"", ""));
-          }
+  protected Map<String, String> readNPIOrgDataStream(InputStream inputStream) throws IOException {
+    Map<String, String> npiProcessedData = new HashMap<String, String>();
+
+    try (final InputStream npiStream = inputStream;
+        final BufferedReader npiReader = new BufferedReader(new InputStreamReader(npiStream))) {
+
+      String line = "";
+      while ((line = npiReader.readLine()) != null) {
+        String npiDataColumns[] = line.split("\t");
+        if (npiDataColumns.length == 2) {
+          npiProcessedData.put(
+              npiDataColumns[0].replace("\"", ""), npiDataColumns[1].replace("\"", ""));
         }
-
-      } catch (IOException e) {
-        LOGGER.error("Unable to read NPI data:" + e.getMessage());
-        throw new IOException(e);
       }
-    } else {
-      LOGGER.error("Unable to read stream for NPI Resource");
-      throw new IOException("Unable to read stream for NPI Resource");
-    }
 
-    if (includeFakeNPIOrgCode) {
-      npiOrgHashMap.put(FAKE_NPI_NUMBER, FAKE_NPI_ORG_NAME);
+    } catch (IOException e) {
+      LOGGER.error("Unable to read NPI data:" + e.getMessage());
+      throw new IOException(e);
```

Comment:
I don't think we've ever set this rule amongst our team, but I've always seen the practice of "log or throw, but dont do both". It cuts down on log noise I guess, and we'll have the stacktrace if the exception is thrown.

Reply:
Done